builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -876,3 +878,68 @@ func protocolForValue(v string) string {
 	log.Printf("[WARN] Unable to determine valid protocol: no matching protocols found")
 	return protocol
 }
+
+func getLambdaENIs(conn *ec2.EC2, d *schema.ResourceData) []ec2.NetworkInterface {
+	filter := &ec2.Filter{
+		Name:   aws.String("group-id"),
+		Values: []*string{aws.String(d.Id())},
+	}
+	params := &ec2.DescribeNetworkInterfacesInput{
+		Filters: []*ec2.Filter{filter},
+	}
+	networkInterfaceResp, err := conn.DescribeNetworkInterfaces(params)
+
+	lambdaENIs := make([]ec2.NetworkInterface, 0)
+	if err != nil {
+		log.Printf("[WARN] Unable to fetch ENIs: %s", err)
+		return lambdaENIs
+	}
+
+	v := networkInterfaceResp.NetworkInterfaces
+	for _, eni := range v {
+		if strings.Contains(*eni.Description, "AWS Lambda VPC ENI") {
```

Comment:
^ Theoretically we could still delete ENI which was originally created via IAM role that was called `awslambda_...`, but that would happen less likely than if we did a simple SG & `description` match.